File: Prixly/Prixly.Dal/Models/StandardRewardRuleDto.cs
using Prixly.Dal.Enums;

namespace Prixly.Dal.Models
{
    public class StandardRewardRuleDto
    {
        public MerchantCategoryCode Category { get; set; }
        public BonusType BonusType { get; set; } // Points, Miles, Cashback
        public decimal Multiplier { get; set; }  // e.g., 3.0 for 3x points
    }
}

File: Prixly/Prixly.Client/src/models/standard-reward-rule-dto.ts
import { BonusType } from "./enums/bonus-type";
import { MerchantCategoryCode } from "./enums/merchant-category-code";

export interface StandardRewardRuleDto {
  category: MerchantCategoryCode;
  bonusType: BonusType;
  multiplier: number;
}
